Add local address binding option for clients

PeterHajdu opened the pull request:
In case the client has multiple ip addresses it can define a preferred source address to initiate requests from.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio-http?pullRequest=1868) <br/>All committers have signed the CLA.

PeterHajdu commented:
Hm...  For some reason I'm unable to sign the CLA.  It keeps asking me to sign in even though I'm already signed in...

jdegoes commented:
@PeterHajdu There's usually a mismatch between your signed email and your Github email.

PeterHajdu committed:
Add local address binding option for clients

In case the client has multiple ip addresses it can define a preferred
source address to initiate requests from.

PeterHajdu commented:
@jdegoes  Good to know.  Looks fine now.

PeterHajdu commented:
Hello, please let me know if I can help somehow to proceed with this.  I'm not sure if I still need to do something before a review.

vigoo commented:
Thanks!
